lib/IDE/CodeCompletion.cpp

```diff
@@ -2481,16 +2481,17 @@ class CompletionLookup final : public swift::VisibleDeclConsumer {
         // What's left is the result type.
         if (ResultType->isVoid()) {
           OS << "Void";
-        } else if (!IsImplicitlyCurriedInstanceMethod
-                   && FD->getAttrs().hasAttribute<ImplicitlyUnwrappedOptionalAttr>()) {
+        } else {
           // As we did with parameters in addParamPatternFromFunction,
           // for regular methods we'll print '!' after implicitly
           // unwrapped optional results.
-          auto ObjectType = ResultType->getOptionalObjectType();
-          OS << ObjectType->getStringAsComponent();
-          OS << "!";
-        } else {
-          ResultType.print(OS);
+          bool IsIUO =
+              !IsImplicitlyCurriedInstanceMethod &&
+              FD->getAttrs().hasAttribute<ImplicitlyUnwrappedOptionalAttr>();
+
+          PrintOptions PO;
+          PO.PrintOptionalAsImplicitlyUnwrapped = IsIUO;
+          ResultType.print(OS, PO);
```

Comment:
This fix is going to cause other problems. All optionals will be printed with `!`, so if e.g. if you try your change with a function that returns `[Int?]!`, I think you'll find it prints as `[Int!]!`.

This applies to the earlier fix for parameters as well.

Reply:
I don't think so. This option only applies to the first Optional found.
https://github.com/apple/swift/blob/master/lib/AST/ASTPrinter.cpp#L3748-L3754

I will add a test case for it.

Reply:
Ah yes, you're right, I had forgotten the code ensured that.